In [3]:
# Step 1 : Building Two Datasets With Matching Means

branch_A = [98, 100, 102, 99, 106, 100, 100]
branch_B = [50, 150, 74, 120, 100, 130, 70]

branch_A_mean = sum(branch_A) / len(branch_A)
branch_B_mean = sum(branch_B) / len(branch_B)

print("Mean A :", branch_A_mean)
print("Mean B :", branch_B_mean)

Mean A : 100.71428571428571
Mean B : 99.14285714285714


In [4]:
# Step 2: Calculating the Range for Both

range_A = max(branch_A) - min(branch_A)
range_B = max(branch_B) - min(branch_B)

print("Branch A Range:", range_A)
print("Branch B Range:", range_B)

Branch A Range: 8
Branch B Range: 100


##### Surprising of how bigger the range of branch B is than branch A 

In [7]:
# Step 3: Calculate Variance by Hand
 # A
squared_A = []
for value in branch_A:
    distance = value - branch_A_mean
    squared_distance = distance ** 2
    squared_A.append(squared_distance)

total_squared_A = 0
for value in squared_A:
    total_squared_A += value

variance_A = total_squared_A / len(branch_A)
print("Branch A Variance:", variance_A)


 # B
squared_B = []

for value in branch_B:
    distance = value - branch_B_mean
    squared_distance = distance ** 2
    squared_B.append(squared_distance)

total_squared_B = 0
for value in squared_B:
    total_squared_B += value

variance_B = total_squared_B / len(branch_B)
print()
print("Branch B Variance:", variance_B)

Branch A Variance: 5.918367346938775

Branch B Variance: 1124.408163265306


In [ ]:
# Step 4: Calculating Standard Deviation by Hand

###### Taking the square root of variance brings the measurement back to the original units, so that the sales can be measured in FCFA instead of FCFA squared.

In [9]:
std_A = variance_A ** 0.5
std_B = variance_B ** 0.5

print("Branch A Standard Deviation:", std_A)
print()
print("Branch B Standard Deviation:", std_B)

Branch A Standard Deviation: 2.4327694808466287

Branch B Standard Deviation: 33.53219592071635


In [14]:
# Step 5: Comparing All Six Numbers Side by Side

print("-----Summary-----")
print("Branch_A")
print("- Mean: ", branch_A_mean)
print("- Range: ", range_A)
print("- Variance: ", variance_A)
print("- Standard deviation: ", std_A)

print()
print("Branch_B")
print("- Mean: ", branch_B_mean)
print("- Range: ", range_B)
print("- Variance: ", variance_B)
print("- Standard deviation: ", std_B)

-----Summary-----
Branch_A
- Mean:  100.71428571428571
- Range:  8
- Variance:  5.918367346938775
- Standard deviation:  2.4327694808466287

Branch_B
- Mean:  99.14285714285714
- Range:  100
- Variance:  1124.408163265306
- Standard deviation:  33.53219592071635


##### The branches have a close average, but Branch B is dramatically more unpredictable.

In [ ]:
# Step 6: Writing the Business-Relevant Interpretation

##### Branch A is more predictable.

##### Branch A has a very small standard deviation of about 2.43, meaning its daily sales usually stay very close to the average of 100.71. Therefore, the average is a reliable guide for planning.

##### Branch B has a much larger standard deviation of about 33.53. Its sales change from day to day. A manager who plans only according to the average could be unprepared on days when sales are much higher than expected and could have too many resources on days when sales are much lower.

##### For example, if both branches were staffed based only on their average sales of 100, Branch B could experience understaffing on a high-sales day and overstaffing on a low-sales day.

In [ ]:
# Step 7: 

##### I would keep inventory levels more stable at Branch A because its sales are predictable and usually close to the average. For Branch B, I would keep additional safety stock because its sales fluctuate, making stockouts more likely on high-sales days. I would also monitor Branch B's sales more frequently instead of relying only on its average.